sktime/proba/t.py

```diff
@@ -0,0 +1,164 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Student's t-distribution."""
+
+__author__ = ["Alex-JG3"]
+
+import numpy as np
+import pandas as pd
+from scipy.special import betaincinv, gamma, hyp2f1, loggamma
+
+from sktime.proba.base import BaseDistribution
+
+
+class TDistribution(BaseDistribution):
+    """Student's t-distribution (sktime native).
+
+    Parameters
+    ----------
+    mean : float or array of float (1D or 2D)
+        mean of the normal distribution
+    sd : float or array of float (1D or 2D), must be positive
+        standard deviation of the normal distribution
+    df : float or array of float (1D or 2D), must be positive
+        Degrees of freedom of the t-distribution
+    index : pd.Index, optional, default = RangeIndex
+    columns : pd.Index, optional, default = RangeIndex
+
+    Example
+    -------
+    >>> from sktime.proba.t import TDistribution
+
+    >>> n = TDistribution(mu=[[0, 1], [2, 3], [4, 5]], sigma=1, df=10)
+    """
+
+    _tags = {
+        "capabilities:approx": ["pdfnorm", "energy"],
+        "capabilities:exact": ["mean", "var", "pdf", "log_pdf", "cdf", "ppf"],
+        "distr:measuretype": "continuous",
+    }
+
+    def __init__(self, mu, sigma, df=2, index=None, columns=None):
```

Comment:
@fkiraly, I have change the default to `df` to 2. This is because the variance tests were failing because [`_check_output__format`](https://github.com/sktime/sktime/blob/main/sktime/proba/tests/test_all_distrs.py#L143) checks to see if all variances are greater than or equal to zero. Variance for the t-distribution is undefined when the degrees of freedom is 1 or less. Do you think this test needs to be updated?

Reply:
I see, interesting.
Perhaps we need to adapt the tests.

Although, are you sure the implementation is correct? I thought even for `df=2` the variance was infinite, so there might be something not right.

For `df=1`, the mean is undefined.

One thought: can we allow nans in the test?

Reply:
>Although, are you sure the implementation is correct? I thought even for df=2 the variance was infinite, so there might be something not right.

Here is an example for the current implementation.

```python
>>> import pandas as pd
>>> from sktime.proba.t import TDistribution
>>> d = TDistribution(mu=0, sigma=1, df=[[0, 1], [2, 3], [4, 5]])
>>> x = pd.DataFrame([[0.0, 1.0], [1.5, 2.0], [3.0, 5.0]])
>>> d.var()
     0         1
0  NaN       NaN
1  inf  3.000000
2  2.0  1.666667
```
Does this look right to you?

>For df=1, the mean is undefined.

Currently that is not happening. I will make the change.

>One thought: can we allow nans in the test?

Could we use `np.greater_equal` and use the `where` argument to ignore `nans`?

Reply:
> Does this look right to you?

Hmmm, yes, though shouldn't the first row not also be `inf`?
Also, what were you using `x` for? (were you going to write another line?)

> Could we use `np.greater_equal` and use the `where` argument to ignore nans?

That would work, although replacing nan-s with `np.inf` might be a more principled solution? All univariate distributions have a variance, if you allow for infinity (to mean divergence)